src/coreclr/jit/compiler.cpp

```diff
@@ -3195,6 +3195,10 @@ void Compiler::compInitOptions(JitFlags* jitFlags)
 
     opts.compProcedureSplitting = jitFlags->IsSet(JitFlags::JIT_FLAG_PROCSPLIT);
 
+#ifdef DEBUG
```

Comment:
I assume JitFakeProcedureSplitting would only be used to test the JIT's support for hot/cold splitting -- we likely wouldn't want it enabled outside of testing, since it shouldn't improve performance (and may actually degrade it due to increased memory usage, and an inability to pack hot sections together). Thus, I decided to make it available only on Debug/Checked builds, though I'm curious to hear Bruce's thoughts on this as well.

Reply:
I mean something like this, in `compiler.h` or somewhere else appropriate.

```
#ifdef DEBUG
#define FAKE_HOT_COLD_SPLITTING
#endif
```

Then in various places, you can use `#define FAKE_HOT_COLD_SPLITTING` instead of `#ifdef DEBUG`. This makes your code more clear when your intention is to check if `FAKE_HOT_COLD_SPLITTING` is enabled or not, and you can also turn off just `FAKE_HOT_COLD_SPLITTING` (obviously, by commenting out just that line), not the entire `DEBUG` mode.

Reply:
The question is whether we want to use/test fake splitting in Release builds as well as Debug/Checked (note that most JIT testing is done with Checked, which defines DEBUG). I think it's fine for now to keep it under DEBUG. If we want it in Release builds, we'll simply define JitFakeProcedureSplitting() in release builds, then move uses out of the DEBUG sections.

Reply:
I see, sorry for the confusion. Bruce and I synced up, and he thinks wrapping the fake splitting functionality with `DEBUG` is fine for now -- the CI pipeline that will test fake/stress splitting uses Checked builds, but I will keep this suggestion in mind if we decide to run fake/stress splitting in Release later.

Thank you for taking a look at my PR!

Reply:
Sounds fine to me.